src/factoid.rs

```diff
@@ -77,40 +77,45 @@ impl FactoidKnowledge for Brain {
 }
 
 // e.g. awoo -> awooooo or meow -> meoooow
-fn is_extension(base: &mut Iterator<Item=char>, mut candidate: &mut Iterator<Item=char>)  -> bool {
-    let h = match base.next() {
-        Some(chr) => chr,
-        None => {
-            // Base is out of characters, is this good or bad?
-            return candidate.next() == None;
-        }
-    };
+fn is_extension(base: &'static str, candidate: &str) -> bool {
+    fn inner_ext_check(base: &mut Iterator<Item=char>, mut candidate: &mut Iterator<Item=char>)  -> bool {
+        let h = match base.next() {
+            Some(chr) => chr,
+            None => {
+                // Base is out of characters, is this good or bad?
```

Comment:
ME: please show me Some chars in the order that they were made 
COMPUTER: thats too hard. heres None chars i think are good. Do you like this